src/huggingface_hub/hf_api.py

```diff
@@ -1094,82 +1127,80 @@ def model_info(
         return ModelInfo(**d)
 
     @_deprecate_positional_args
-    def list_repo_files(
+    def dataset_info(
         self,
         repo_id: str,
         *,
         revision: Optional[str] = None,
-        repo_type: Optional[str] = None,
         token: Optional[str] = None,
         timeout: Optional[float] = None,
-    ) -> List[str]:
+    ) -> DatasetInfo:
         """
-        Get the list of files in a given repo.
+        Get info on one specific dataset on huggingface.co
+
+        Dataset can be private if you pass an acceptable token.
 
         Args:
             repo_id (`str`):
                 A namespace (user or an organization) and a repo name separated
                 by a `/`.
             revision (`str`, *optional*):
-                The revision of the model repository from which to get the
+                The revision of the dataset repository from which to get the
                 information.
-            repo_type (`str`, *optional*):
-                Set to `"dataset"` or `"space"` if uploading to a dataset or
-                space, `None` or `"model"` if uploading to a model. Default is
-                `None`.
             token (`str`, *optional*):
                 An authentication token [1]_.
             timeout (`float`, *optional*):
                 Whether to set a timeout for the request to the Hub.
 
         Returns:
-            `List[str]`: the list of files in a given repository.
+            [`DatasetInfo`]: The dataset repository information.
 
         References:
 
         - [1] https://huggingface.co/settings/tokens
         """
-        if repo_type is None or repo_type == "model":
-            info = self.model_info(
-                repo_id=repo_id, revision=revision, token=token, timeout=timeout
-            )
-        elif repo_type == "dataset":
-            info = self.dataset_info(
-                repo_id=repo_id, revision=revision, token=token, timeout=timeout
-            )
-        else:
-            raise ValueError("Spaces are not available yet.")
+        if token is None:
+            token = HfFolder.get_token()
 
-        return [f.rfilename for f in info.siblings]
+        path = (
+            f"{self.endpoint}/api/datasets/{repo_id}"
+            if revision is None
+            else f"{self.endpoint}/api/datasets/{repo_id}/revision/{revision}"
+        )
+        headers = {"authorization": f"Bearer {token}"} if token is not None else None
+        r = requests.get(path, headers=headers, timeout=timeout)
+        r.raise_for_status()
+        d = r.json()
```

Comment:
do we need to do the same as we did in https://github.com/huggingface/huggingface_hub/pull/783 here? (we could have a utility function to avoid repeating code if that's the case)

Reply:
I'm not sure that's the case, IIUC it was mostly linked to `delete_repo` having different errors when they should have been the same?